### DISCUSSION TITLE
SER模型可以使用onnx推理吗？

### DISCUSSION THREAD

**Discussion by TOMORINAONAO** [PaddlePaddle/PaddleOCR#13175] [Q&A]
自己训练的ser模型，实际应用环境可能没法使用paddle框架，所以想用onnx推理。
我尝试使用paddle2onnx，设置opset=15成功将ser模型转为onnx模型没有报错。现在有两个问题：
1.paddleocr这边的文档和issue都差不多看了，没看到ser模型onnx推理的相关内容，这个可行吗？要是有教程就好了。
2.我看读取图片后要用一个nlp模型将ocr模型的识别结果编码，paddlenlp我没用过，这个nlp模型也能转为onnx模型加进代码吗？

**Comment by GreatV** [comment]
你可能得修改一下代码 

https://github.com/PaddlePaddle/PaddleOCR/blob/main/ppstructure/kie/README_ch.md#423-%E5%9F%BA%E4%BA%8Epaddleinference%E7%9A%84%E9%A2%84%E6%B5%8B

**Comment by SWHL** [comment]
ans问题1: 这个可以自己debug尝试一下，没有那么详细的教程。大概思路是将送入paddle模型前的变量，转成numpy格式，送入onnx模型中，查看推理得出的数值与经过paddle模型推理得出的数值是否误差较大，从而确定是否可行。
ans问题2：nlp模型需要自行尝试转换，有一定几率不行，可能存在不支持算子
